src/python/pants/backend/awslambda/python/target_types.py

```diff
@@ -20,23 +20,28 @@
     InjectDependenciesRequest,
     InjectedDependencies,
     InvalidFieldException,
+    SecondaryOwnerMixin,
     StringField,
     Target,
     WrappedTarget,
 )
 from pants.engine.unions import UnionRule
+from pants.source.filespec import Filespec
 from pants.source.source_root import SourceRoot, SourceRootRequest
 
 
 class PythonAwsLambdaSources(PythonSources):
     expected_num_files = range(0, 2)
 
 
-class PythonAwsLambdaHandlerField(StringField, AsyncFieldMixin):
+class PythonAwsLambdaHandlerField(StringField, AsyncFieldMixin, SecondaryOwnerMixin):
     """Entry point to the AWS Lambda handler.
 
     You can specify a full module like 'path.to.module:handler_func' or use a shorthand to specify a
     file name, using the same syntax as the `sources` field, e.g. 'lambda.py:handler_func'.
+
+    You must use the file name shorthand for file arguments and `--changed-since` to work with this
```

Comment:
I don't quite understand this comment. Do you mean "*for* `--changed-since` to work with this target? That's a bit of a gotcha if so, but probably OK for now.